### DISCUSSION TITLE
VIPS internship / summer of code

### DISCUSSION THREAD

**Discussion by kmartinez** [libvips/libvips#2253] [Help wanted]
If you are a student or work part time on vips - you can join the team as a paid-intern.
We are looking for anyone experienced with vips who can work on an interesting or experimental area - or boost something like docs/examples/help.

Some previous experience programming in C and/or C++ would be highly beneficial, as would any experience of Rust.

We don't have loads of funding - but are keen to help you out with a small payment ($500-2500) depending on the time you commit. This is especially helpful if you are a student! You could even just do a couple of hours a week. Possible areas include:

Smaller (GSoC ~90 hours):

- Improve JPEG-XL animation support
- User JPEG-XL streaming API (encoder)
- we support C++ plugins already, adding support for Rust plugins would be great
- extend fuzzing to a wider range of loaders, eg. openslide and libdicom
- link to tensorflow / pytorch / etc. 
- tutorial materials, examples

Larger (GSoC ~175 hours):

- GPU acceleration
- a better Rust binding
- a better Go binding

See https://github.com/libvips/libvips/discussions/categories/ideas for more enhancement ideas

but anything helpful will be considered. Don't hesitate to contact us to discuss ideas - offline or here. The dev team will select people depending on the budget we have available.

**Comment by sethify** [comment]
@kmartinez the budget has been enhanced 👍

**Reply by kmartinez** [reply]
We really appreciate all the support!

**Comment by HarshCasper** [comment]
Hi,

My name is Harsh and I'm a Senior Year Undergraduate student from Sathyabama Institute of Science and Technology, India. I am a [Google Summer of Code](https://summerofcode.withgoogle.com/) 2021 Student Developer at [MetaCall](http://metacall.io/) and a [Google Season of Docs](https://developers.google.com/season-of-docs) Technical Writer at [moja global](http://moja.global/). In the past, I have worked at [Red Hat](http://www.redhat.com/), [Qxf2](https://qxf2.com/) and [Major League Hacking](https://fellowship.mlh.io/).

I would like to work on developing the tutorial/examples for libvips to boost the content outreach for the project and help lower the bar of entry for new contributors and users. My experience in documentation engineering and content development comes in handy to undertake this project and I can put forward a plan/proposal for the same. I would collaborate with the developers at libvips to iterate on the same and make sure that our goals and deliverables are put on the table before I start my work.

My idea revolves around setting up a website using modern static site generators to showcase all our documentation pieces through a full code-complete solution, with documentation, guidelines and more. I have considerable experience in Docusaurus, Sphinx, Hugo, Jekyll and would be able to utilize the same to meet the goals and the deliverables with a certain finish.

I would be available from the entire Fall season, beginning from September 5th. I would be able to commit 20-25 hours per week to the project and would be able to plan the deliverables in a manner that would allow me to be consistent and fruitful in my efforts, and would be able to make a real impact.

I have reached out to @kmartinez with my overall interest in the project and a resume highlighting all my past experiences and background. I would love to discuss the same with the Developer team so that we can further iterate on possibilities before we kick-start the project. 

Looking forward to a positive response!
